lib/types/swagger_1.js

```diff
@@ -85,8 +85,10 @@ Swagger1.prototype.resolveSubResource = function (resource) {
     if (this.spec.basePath)
       baseUrl = URI(this.spec.basePath).absoluteTo(baseUrl);
 
-    if (this.getFormatVersion() === '1.0' && baseUrl.suffix() === 'json')
+    if (baseUrl.suffix() === 'json') {
       baseUrl.filename('');
+      if (resource.indexOf('.json') === -1) resource += '.json';
+    }
```

Comment:
Ok I understand now. 
`if (resource.indexOf('.json') === -1) resource += '.json';` definitely no, it's not a real case, you fixing problem inside tests just add `.json` inside swagger file.
About `this.getFormatVersion() === '1.0'` there similar PR for `swagger.js` https://github.com/swagger-api/swagger-js/pull/638
But in really I spend a lot of time fine tuning this function, so I don't want to change anything without real use case.
I remember that initially I also removed version check but found few APIs that was broken because of that.
Tomorrow I will retest to be sure.